src/metabase/db/custom_migrations.clj

```diff
@@ -205,9 +205,40 @@
                                  :set    {:visualization_settings visualization_settings}
                                  :where  [:= :id id]}))]
     (run! update! (eduction (keep (fn [{:keys [id visualization_settings]}]
-                                    (let [updated (update-legacy-field-refs visualization_settings)]
+                                    (let [updated (update-legacy-field-refs-in-viz-settings visualization_settings)]
                                       (when (not= visualization_settings updated)
                                         {:id                     id
                                          :visualization_settings updated}))))
                             (t2/reducible-query {:select [:id :visualization_settings]
                                                  :from   [:report_card]})))))
+
+(defn- update-legacy-field-refs-in-result-metadata [result-metadata]
+  (let [old-to-new (fn [ref]
+                     (match ref
```

Comment:
Note this is exactly the same as the previous migration https://github.com/metabase/metabase/pull/30787/files#diff-be85115f22d8821f0a204e2c870538dc80f6c4191640e6f02c7cc558dbda6544R179